e2e/tests-dfx/wallet.bash

```diff
@@ -135,39 +133,41 @@ teardown() {
     dfx_new
     dfx_start
     WALLET_ID=$(dfx identity get-wallet)
-    CREATE_RES=$(dfx canister --no-wallet call "${WALLET_ID}" wallet_create_wallet "(record { cycles = (2000000000000:nat64); settings = record {controller = opt principal \"$(dfx identity get-principal)\";};})")
+    CREATE_RES=$(dfx canister call "${WALLET_ID}" wallet_create_wallet "(record { cycles = (2000000000000:nat64); settings = record {controller = opt principal \"$(dfx identity get-principal)\";};})")
     CHILD_ID=$(echo "${CREATE_RES}" | tr '\n' ' ' |  cut -d'"' -f 2)
-    assert_command dfx canister --no-wallet call "${CHILD_ID}" wallet_balance '()'
+    assert_command dfx canister call "${CHILD_ID}" wallet_balance '()'
 }
 
-@test "bypass wallet call as user" {
+@test "forward user call through wallet" {
     dfx_new
     install_asset identity
     dfx_start
-    assert_command dfx canister --no-wallet create --all
+    WALLET=$(dfx identity get-wallet)
+    assert_command dfx canister --wallet "$WALLET" create --all
     assert_command dfx build
-    assert_command dfx canister --no-wallet install --all
+    assert_command dfx canister --wallet "$WALLET" install --all
 
-    CALL_RES=$(dfx canister --no-wallet call e2e_project fromCall)
+    CALL_RES=$(dfx canister --wallet "$WALLET" call e2e_project fromCall)
     CALLER=$(echo "${CALL_RES}" | cut -d'"' -f 2)
-    ID=$(dfx identity get-principal)
-    assert_eq "$CALLER" "$ID"
+    assert_eq "$CALLER" "$WALLET"
 
-    assert_command dfx canister --no-wallet call e2e_project amInitializer
-    assert_eq '(true)'
+    assert_command dfx canister call "$WALLET" wallet_call \
+        "(record { canister = principal \"$(dfx canister id e2e_project)\"; method_name = \"amInitializer\"; args = blob \"DIDL\00\00\"; cycles = (0:nat64)})"
+    assert_eq '(variant { 17_724 = record { 153_986_224 = blob "DIDL\00\01~\01" } })'  # True in DIDL.
 }
 
-@test "bypass wallet call as user: deploy" {
+@test "forward user call through wallet: deploy" {
     dfx_new
     install_asset identity
     dfx_start
-    assert_command dfx deploy --no-wallet
-
-    CALL_RES=$(dfx canister --no-wallet call e2e_project fromCall)
+    WALLET=$(dfx identity get-wallet)
+    assert_command dfx deploy --wallet "$WALLET"
+    CALL_RES=$(dfx canister --wallet "$WALLET" call e2e_project fromCall)
     CALLER=$(echo "${CALL_RES}" | cut -d'"' -f 2)
-    ID=$(dfx identity get-principal)
-    assert_eq "$CALLER" "$ID"
+    assert_eq "$CALLER" "$WALLET"
 
-    assert_command dfx canister --no-wallet call e2e_project amInitializer
-    assert_eq '(true)'
```

Comment:
Can we retain this test (`dfx deploy` vs `dfx deploy --wallet "$WALLET"`, then the rest)

Reply:
As in, not changing the --wallet flag in the `canister call` call?